dataland-frontend/src/App.vue

```diff
@@ -17,116 +18,104 @@ import SessionDialog from "@/components/general/SessionDialog.vue";
 import { KEYCLOAK_INIT_OPTIONS } from "@/utils/Constants";
 import { useSharedSessionStateStore } from "@/stores/Stores";
 
-export default defineComponent({
-  name: "app",
-  components: { DynamicDialog },
-
-  data() {
-    return {
-      keycloakPromise: undefined as undefined | Promise<Keycloak>,
-      resolvedKeycloakPromise: undefined as undefined | Keycloak,
-      keycloakAuthenticated: false,
-      functionIdOfSessionSetInterval: undefined as number | undefined,
-    };
-  },
+const sharedStore = useSharedSessionStateStore();
+const keycloakPromise = ref<Promise<Keycloak> | undefined>();
+const resolvedKeycloakPromise = ref<Keycloak | undefined>();
+const keycloakAuthenticated = ref(false);
+const functionIdOfSessionSetInterval = ref<number | undefined>();
+const dialog = useDialog();
 
-  watch: {
-    currentRefreshTokenInSharedStore(newRefreshToken: string) {
-      if (this.resolvedKeycloakPromise && newRefreshToken) {
-        this.resolvedKeycloakPromise.refreshToken = newRefreshToken;
-        clearInterval(this.functionIdOfSessionSetInterval);
-        const openSessionWarningModalBound = this.openSessionWarningModal.bind(this);
-        this.functionIdOfSessionSetInterval = startSessionSetIntervalFunctionAndReturnItsId(
-          this.resolvedKeycloakPromise,
-          openSessionWarningModalBound,
-        );
-      }
-    },
-  },
+const currentRefreshTokenInSharedStore = computed(() => sharedStore.refreshToken);
 
-  computed: {
-    currentRefreshTokenInSharedStore() {
-      return useSharedSessionStateStore().refreshToken;
-    },
-  },
+watch(currentRefreshTokenInSharedStore, (newRefreshToken) => {
+  if (typeof newRefreshToken === "string") {
```

Comment:
else throw an exception maybe? Such that the case is not unhandled.

Reply:
But you assumed a type now.

Reply:
In Vue, the watch function is used for side-effects and it reacts to changes in reactive properties. When a watched property changes, the function is executed. The variable passed to this function (newRefreshToken) will contain the new value of the watched property.

TypeScript doesn't know the runtime type of this variable out-of-the-box. While I might define the reactive property as a string, Vue's reactivity system won't enforce that it stays a string at runtime.

Therefore, checking if (typeof newRefreshToken === "string") confirms that the new value is indeed a string before proceeding with the rest of the logic inside the watch callback. This makes the code more robust against unexpected data types.

Reply:
My bad. I just did not realize that `sharedStore.refreshToken` had this type.